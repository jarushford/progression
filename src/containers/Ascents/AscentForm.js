import React, { Component } from 'react'

class AscentForm extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      location: '',
      caption: '',
      grade: '',
      ascentAdded: false
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    console.log(e.target.value)
    this.setState({ [name]: value })
  }

  handleSubmit = (e) => {
    e.preventDefault()

  }

  render() {
    const { name, location, caption, grade } = this.state
    return (
      <form className="ascent-form" onSubmit={this.handleSubmit}>
        <h2>ADD ASCENT</h2>
        <input name="name"
          value={name}
          onChange={this.handleChange}
          placeholder="name"
          autoFocus={true}
        />
        <input name="location"
          value={location}
          onChange={this.handleChange}
          placeholder="location"
        />
        <select id="grade"
          name="grade"
          value={grade}
          onChange={this.handleChange}
        >
          <option value="null"> -- </option>
          <option value="0"> V0 / 4 </option>
          <option value="1"> V1 / 5 </option>
          <option value="2"> V2 / 5+ </option>
          <option value="3"> V3 / 6A </option>
          <option value="4"> V4 / 6B </option>
          <option value="5"> V5 / 6C </option>
          <option value="6"> V6 / 7A </option>
          <option value="7"> V7 / 7A+ </option>
          <option value="8"> V8 / 7B+ </option>
          <option value="9"> V9 / 7C </option>
          <option value="10"> V10 / 7C+ </option>
          <option value="11"> V11 / 8A </option>
          <option value="12"> V12 / 8A+ </option>
          <option value="13"> V13 / 8B </option>
          <option value="14"> V14 / 8B+ </option>
          <option value="15"> V15 / 8C </option>
          <option value="16"> V16 / 8C+ </option>
        </select>
        <textarea name="caption"
          value={caption}
          onChange={this.handleChange}
          placeholder="caption"
        />
        <button>Add Ascent</button>
      </form>
    )
  }
}

export default AscentForm

