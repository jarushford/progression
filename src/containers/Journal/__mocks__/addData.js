export const addDataThunk = jest.fn()