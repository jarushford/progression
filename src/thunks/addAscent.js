import { fetchAscentsThunk } from './fetchAscents'
import { clearAscents, setError } from '../actions'

export const addAscentThunk = (ascent) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/ascents/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(ascent),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add ascent')
      }
      await dispatch(clearAscents())
      dispatch(fetchAscentsThunk(ascent.user_id))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}