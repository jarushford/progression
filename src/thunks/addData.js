import { clearThunkHelper } from '../utils/thunkHelpers'
import { fetchDataThunk } from './fetchData'
import { setError } from '../actions'

export const addDataThunk = (data, type) => {
  return async (dispatch) => {
    try {
      const url = `http://localhost:3000/api/progressionusers/${type}s/new`
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(data),
        headers: {
          "Content-Type": "application/json"
        }
      })

      if (!response.ok) {
        throw Error(`Could not add ${type}`)
      }

      await dispatch(clearThunkHelper(type)())
      type === 'journal' || type === 'milestone'
        ? dispatch(fetchDataThunk(data.user_id, data.project_id, type))
        : dispatch(fetchDataThunk(data.user_id, '', type))

      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}