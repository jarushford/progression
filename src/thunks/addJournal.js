import { fetchJournalThunk } from './fetchJournal'
import { clearJournal, setError } from '../actions'

export const addJournalThunk = (entry) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/journal/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(entry),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add entry')
      }
      await dispatch(clearJournal())
      dispatch(fetchJournalThunk(entry.user_id, entry.project_id))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}