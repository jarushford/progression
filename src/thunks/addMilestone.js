import { fetchMilestonesThunk } from './fetchMilestones'
import { clearMilestones, setError } from '../actions'

export const addMilestoneThunk = (milestone) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/milestones/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(milestone),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add project')
      }
      await dispatch(clearMilestones())
      dispatch(fetchMilestonesThunk(milestone.user_id, milestone.project_id))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}