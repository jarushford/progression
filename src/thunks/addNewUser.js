import { setError } from '../actions'
import { loginUserThunk } from './loginUser'

export const addNewUserThunk = (user) => {
  return async (dispatch) => {
    try {
      const url = `https://progression-backend.herokuapp.com/api/progressionusers/new`
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(user),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw new Error('User already exists, try another email!')
      }
      await dispatch(loginUserThunk(user))
      return true
    } catch (error) {
      dispatch(setError(error.message))
      return false
    }
  }
}