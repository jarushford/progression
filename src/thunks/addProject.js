import { fetchProjectsThunk } from './fetchProjects'
import { clearProjects } from '../actions'

export const addProjectThunk = (project) => {
  return async (dispatch) => {
    try {
      const url = 'http://localhost:3000/api/progressionusers/projects/new'
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(project),
        headers: {
          "Content-Type": "application/json"
        }
      })
      if (!response.ok) {
        throw Error('Could not add project')
      }
      await dispatch(clearProjects())
      dispatch(fetchProjectsThunk(project.user_id))
    } catch (error) {
      console.log(error)
    }
  }
}