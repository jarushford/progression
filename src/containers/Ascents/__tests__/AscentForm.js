import React from 'react'
import { shallow } from 'enzyme'
import { addDataThunk } from '../../../thunks/addData'
import { AscentForm, mapStateToProps, mapDispatchToProps } from '../AscentForm'

jest.mock('../../../thunks/addData')

describe('AscentForm', () => {
  let mockUser
  let mockAddAscent
  
  beforeEach(() => {
    mockUser = {
      name: 'T-Rex',
      email: 'chompy@yahoo.com',
      password: 'smallarms',
      id: 4
    }
    mockAddAscent = jest.fn().mockImplementation(() => true)
  })

  describe('AscentForm Component', () => {
    it('should match the snapshot for boulder', () => {
      const wrapper = shallow(<AscentForm user={mockUser} addAscent={mockAddAscent} error='' disciplineBoulder={true} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot for sport', () => {
      const wrapper = shallow(<AscentForm user={mockUser} addAscent={mockAddAscent} error='' disciplineBoulder={false} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the correct default local state', () => {
      const expected = {
        name: '',
        location: '',
        caption: '',
        grade: '',
        ascentAdded: false
      }
      const wrapper = shallow(<AscentForm user={mockUser} addAscent={mockAddAscent} error='' disciplineBoulder={true} />)

      expect(wrapper.state()).toEqual(expected)
    })

    it('should update state onChange of the inputs', () => {
      const expected = '7'
      const wrapper = shallow(<AscentForm user={mockUser} addAscent={mockAddAscent} error='' disciplineBoulder={true} />)
      const e = { target: { name: 'grade', value: '7' } }

      wrapper.find('#grade').simulate('change', e)

      expect(wrapper.state().grade).toEqual(expected)
    })

    it('should add an ascent with data from state', async () => {
      const wrapper = shallow(<AscentForm user={mockUser} addAscent={mockAddAscent} error='' disciplineBoulder={true} />)
      const e = { preventDefault: jest.fn() }
      wrapper.setState({
        name: 'Green Lantern',
        location: 'Ute Pass, CO',
        caption: 'Some other words',
        grade: 'V11'
      })

      await wrapper.find('.ascent-form').simulate('submit', e)

      expect(wrapper.state().ascentAdded).toEqual(true)
    })

    it('should not set state if ascent is not added', async () => {
      mockAddAscent = jest.fn().mockImplementation(() => false)
      const wrapper = shallow(<AscentForm user={mockUser} addAscent={mockAddAscent} error='' disciplineBoulder={true} />)
      const e = { preventDefault: jest.fn() }
      wrapper.setState({
        name: 'Green Lantern',
        location: 'Ute Pass, CO',
        caption: 'Some other words',
        grade: 'V11'
      })

      await wrapper.find('.ascent-form').simulate('submit', e)

      expect(wrapper.state().ascentAdded).toEqual(false)
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object with data from the store', () => {
      const mockState = {
        currentUser: mockUser,
        error: ''
      }
      const expected = {
        user: mockUser,
        error: ''
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })

  })

  describe('mapDispatchToProps', () => {
    it('should return a props object with a method addAscent', () => {
      const mockDispatch = jest.fn()
      const expected = addDataThunk({}, 'ascent')

      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.addAscent({}, 'ascent')

      expect(mockDispatch).toBeCalledWith(expected)
    })
  })
})