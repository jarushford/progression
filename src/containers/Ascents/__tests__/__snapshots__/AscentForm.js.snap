// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AscentForm AscentForm Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AscentForm
    addAscent={[MockFunction]}
    disciplineBoulder={true}
    error=""
    user={
      Object {
        "email": "chompy@yahoo.com",
        "id": 4,
        "name": "T-Rex",
        "password": "smallarms",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          ADD ASCENT
        </h2>,
        <input
          autoFocus={true}
          name="name"
          onChange={[Function]}
          placeholder="name"
          value=""
        />,
        <input
          name="location"
          onChange={[Function]}
          placeholder="location"
          value=""
        />,
        <select
          id="grade"
          name="grade"
          onChange={[Function]}
          value=""
        >
          <option
            value="--"
          >
             -- 
          </option>
          <option
            value="0"
          >
             V0 / 4 
          </option>
          <option
            value="1"
          >
             V1 / 5 
          </option>
          <option
            value="2"
          >
             V2 / 5+ 
          </option>
          <option
            value="3"
          >
             V3 / 6A 
          </option>
          <option
            value="4"
          >
             V4 / 6B 
          </option>
          <option
            value="5"
          >
             V5 / 6C 
          </option>
          <option
            value="6"
          >
             V6 / 7A 
          </option>
          <option
            value="7"
          >
             V7 / 7A+ 
          </option>
          <option
            value="8"
          >
             V8 / 7B+ 
          </option>
          <option
            value="9"
          >
             V9 / 7C 
          </option>
          <option
            value="10"
          >
             V10 / 7C+ 
          </option>
          <option
            value="11"
          >
             V11 / 8A 
          </option>
          <option
            value="12"
          >
             V12 / 8A+ 
          </option>
          <option
            value="13"
          >
             V13 / 8B 
          </option>
          <option
            value="14"
          >
             V14 / 8B+ 
          </option>
          <option
            value="15"
          >
             V15 / 8C 
          </option>
          <option
            value="16"
          >
             V16 / 8C+ 
          </option>
        </select>,
        <textarea
          maxLength="500"
          name="caption"
          onChange={[Function]}
          placeholder="caption"
          value=""
        />,
        <button>
          Add Ascent
        </button>,
      ],
      "className": "ascent-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ADD ASCENT",
        },
        "ref": null,
        "rendered": "ADD ASCENT",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoFocus": true,
          "name": "name",
          "onChange": [Function],
          "placeholder": "name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "location",
          "onChange": [Function],
          "placeholder": "location",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="--"
            >
               -- 
            </option>,
            <option
              value="0"
            >
               V0 / 4 
            </option>,
            <option
              value="1"
            >
               V1 / 5 
            </option>,
            <option
              value="2"
            >
               V2 / 5+ 
            </option>,
            <option
              value="3"
            >
               V3 / 6A 
            </option>,
            <option
              value="4"
            >
               V4 / 6B 
            </option>,
            <option
              value="5"
            >
               V5 / 6C 
            </option>,
            <option
              value="6"
            >
               V6 / 7A 
            </option>,
            <option
              value="7"
            >
               V7 / 7A+ 
            </option>,
            <option
              value="8"
            >
               V8 / 7B+ 
            </option>,
            <option
              value="9"
            >
               V9 / 7C 
            </option>,
            <option
              value="10"
            >
               V10 / 7C+ 
            </option>,
            <option
              value="11"
            >
               V11 / 8A 
            </option>,
            <option
              value="12"
            >
               V12 / 8A+ 
            </option>,
            <option
              value="13"
            >
               V13 / 8B 
            </option>,
            <option
              value="14"
            >
               V14 / 8B+ 
            </option>,
            <option
              value="15"
            >
               V15 / 8C 
            </option>,
            <option
              value="16"
            >
               V16 / 8C+ 
            </option>,
          ],
          "id": "grade",
          "name": "grade",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " -- ",
              "value": "--",
            },
            "ref": null,
            "rendered": " -- ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V0 / 4 ",
              "value": "0",
            },
            "ref": null,
            "rendered": " V0 / 4 ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V1 / 5 ",
              "value": "1",
            },
            "ref": null,
            "rendered": " V1 / 5 ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V2 / 5+ ",
              "value": "2",
            },
            "ref": null,
            "rendered": " V2 / 5+ ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V3 / 6A ",
              "value": "3",
            },
            "ref": null,
            "rendered": " V3 / 6A ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V4 / 6B ",
              "value": "4",
            },
            "ref": null,
            "rendered": " V4 / 6B ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V5 / 6C ",
              "value": "5",
            },
            "ref": null,
            "rendered": " V5 / 6C ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V6 / 7A ",
              "value": "6",
            },
            "ref": null,
            "rendered": " V6 / 7A ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V7 / 7A+ ",
              "value": "7",
            },
            "ref": null,
            "rendered": " V7 / 7A+ ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V8 / 7B+ ",
              "value": "8",
            },
            "ref": null,
            "rendered": " V8 / 7B+ ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V9 / 7C ",
              "value": "9",
            },
            "ref": null,
            "rendered": " V9 / 7C ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V10 / 7C+ ",
              "value": "10",
            },
            "ref": null,
            "rendered": " V10 / 7C+ ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V11 / 8A ",
              "value": "11",
            },
            "ref": null,
            "rendered": " V11 / 8A ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V12 / 8A+ ",
              "value": "12",
            },
            "ref": null,
            "rendered": " V12 / 8A+ ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V13 / 8B ",
              "value": "13",
            },
            "ref": null,
            "rendered": " V13 / 8B ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V14 / 8B+ ",
              "value": "14",
            },
            "ref": null,
            "rendered": " V14 / 8B+ ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V15 / 8C ",
              "value": "15",
            },
            "ref": null,
            "rendered": " V15 / 8C ",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " V16 / 8C+ ",
              "value": "16",
            },
            "ref": null,
            "rendered": " V16 / 8C+ ",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "maxLength": "500",
          "name": "caption",
          "onChange": [Function],
          "placeholder": "caption",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Ascent",
        },
        "ref": null,
        "rendered": "Add Ascent",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            ADD ASCENT
          </h2>,
          <input
            autoFocus={true}
            name="name"
            onChange={[Function]}
            placeholder="name"
            value=""
          />,
          <input
            name="location"
            onChange={[Function]}
            placeholder="location"
            value=""
          />,
          <select
            id="grade"
            name="grade"
            onChange={[Function]}
            value=""
          >
            <option
              value="--"
            >
               -- 
            </option>
            <option
              value="0"
            >
               V0 / 4 
            </option>
            <option
              value="1"
            >
               V1 / 5 
            </option>
            <option
              value="2"
            >
               V2 / 5+ 
            </option>
            <option
              value="3"
            >
               V3 / 6A 
            </option>
            <option
              value="4"
            >
               V4 / 6B 
            </option>
            <option
              value="5"
            >
               V5 / 6C 
            </option>
            <option
              value="6"
            >
               V6 / 7A 
            </option>
            <option
              value="7"
            >
               V7 / 7A+ 
            </option>
            <option
              value="8"
            >
               V8 / 7B+ 
            </option>
            <option
              value="9"
            >
               V9 / 7C 
            </option>
            <option
              value="10"
            >
               V10 / 7C+ 
            </option>
            <option
              value="11"
            >
               V11 / 8A 
            </option>
            <option
              value="12"
            >
               V12 / 8A+ 
            </option>
            <option
              value="13"
            >
               V13 / 8B 
            </option>
            <option
              value="14"
            >
               V14 / 8B+ 
            </option>
            <option
              value="15"
            >
               V15 / 8C 
            </option>
            <option
              value="16"
            >
               V16 / 8C+ 
            </option>
          </select>,
          <textarea
            maxLength="500"
            name="caption"
            onChange={[Function]}
            placeholder="caption"
            value=""
          />,
          <button>
            Add Ascent
          </button>,
        ],
        "className": "ascent-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ADD ASCENT",
          },
          "ref": null,
          "rendered": "ADD ASCENT",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": true,
            "name": "name",
            "onChange": [Function],
            "placeholder": "name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "location",
            "onChange": [Function],
            "placeholder": "location",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="--"
              >
                 -- 
              </option>,
              <option
                value="0"
              >
                 V0 / 4 
              </option>,
              <option
                value="1"
              >
                 V1 / 5 
              </option>,
              <option
                value="2"
              >
                 V2 / 5+ 
              </option>,
              <option
                value="3"
              >
                 V3 / 6A 
              </option>,
              <option
                value="4"
              >
                 V4 / 6B 
              </option>,
              <option
                value="5"
              >
                 V5 / 6C 
              </option>,
              <option
                value="6"
              >
                 V6 / 7A 
              </option>,
              <option
                value="7"
              >
                 V7 / 7A+ 
              </option>,
              <option
                value="8"
              >
                 V8 / 7B+ 
              </option>,
              <option
                value="9"
              >
                 V9 / 7C 
              </option>,
              <option
                value="10"
              >
                 V10 / 7C+ 
              </option>,
              <option
                value="11"
              >
                 V11 / 8A 
              </option>,
              <option
                value="12"
              >
                 V12 / 8A+ 
              </option>,
              <option
                value="13"
              >
                 V13 / 8B 
              </option>,
              <option
                value="14"
              >
                 V14 / 8B+ 
              </option>,
              <option
                value="15"
              >
                 V15 / 8C 
              </option>,
              <option
                value="16"
              >
                 V16 / 8C+ 
              </option>,
            ],
            "id": "grade",
            "name": "grade",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " -- ",
                "value": "--",
              },
              "ref": null,
              "rendered": " -- ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V0 / 4 ",
                "value": "0",
              },
              "ref": null,
              "rendered": " V0 / 4 ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V1 / 5 ",
                "value": "1",
              },
              "ref": null,
              "rendered": " V1 / 5 ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V2 / 5+ ",
                "value": "2",
              },
              "ref": null,
              "rendered": " V2 / 5+ ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V3 / 6A ",
                "value": "3",
              },
              "ref": null,
              "rendered": " V3 / 6A ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V4 / 6B ",
                "value": "4",
              },
              "ref": null,
              "rendered": " V4 / 6B ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V5 / 6C ",
                "value": "5",
              },
              "ref": null,
              "rendered": " V5 / 6C ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V6 / 7A ",
                "value": "6",
              },
              "ref": null,
              "rendered": " V6 / 7A ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V7 / 7A+ ",
                "value": "7",
              },
              "ref": null,
              "rendered": " V7 / 7A+ ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V8 / 7B+ ",
                "value": "8",
              },
              "ref": null,
              "rendered": " V8 / 7B+ ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V9 / 7C ",
                "value": "9",
              },
              "ref": null,
              "rendered": " V9 / 7C ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V10 / 7C+ ",
                "value": "10",
              },
              "ref": null,
              "rendered": " V10 / 7C+ ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V11 / 8A ",
                "value": "11",
              },
              "ref": null,
              "rendered": " V11 / 8A ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V12 / 8A+ ",
                "value": "12",
              },
              "ref": null,
              "rendered": " V12 / 8A+ ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V13 / 8B ",
                "value": "13",
              },
              "ref": null,
              "rendered": " V13 / 8B ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V14 / 8B+ ",
                "value": "14",
              },
              "ref": null,
              "rendered": " V14 / 8B+ ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V15 / 8C ",
                "value": "15",
              },
              "ref": null,
              "rendered": " V15 / 8C ",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " V16 / 8C+ ",
                "value": "16",
              },
              "ref": null,
              "rendered": " V16 / 8C+ ",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "maxLength": "500",
            "name": "caption",
            "onChange": [Function],
            "placeholder": "caption",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Ascent",
          },
          "ref": null,
          "rendered": "Add Ascent",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
